### ISSUE TITLE
The `display` property is not being applied to a lot of elements in IE10

### ISSUE THREAD

**Issue by kumarharsh** [FormidableLabs/radium#604]
This is a very weird one I stumbled on today. Apparently, Radium is not applying (or maybe IE10 is dropping) the `display` styles to the element's inline styles. Everything else seems to be working.

I'll try to get a repro on this, maybe tomorrow...

**Comment by kumarharsh** [comment]
I'm using IE11 on Win 10 to *emulate* IE10. Atleast in IE11, things work ok (whatever good that'll do)

**Comment by rynti** [comment]
I'm noticed the exact same issue (missing display properties) when testing my application with Chrome and using the Device Emulator to emulate the iPhone 4.

I've also noticed that it's not only inline styles that are affected: Stylesheets generated for media queries also seem to be missing the display properties. And the issue actually seems to be related to the user agent, I've tried all devices that Chrome offers for emulation and these are the ones that cause the display properties to go missing:

**Amazon Kindle Fire HDX:** `Mozilla/5.0 (Linux; U; en-us; KFAPWI Build/JDQ39) AppleWebKit/535.19 (KHTML, like Gecko) Silk/3.13 Safari/535.19 Silk-Accelerated=true`

**Apple iPhone 4:** `Mozilla/5.0 (iPhone; U; CPU iPhone OS 4_2_1 like Mac OS X; en-us) AppleWebKit/533.17.9 (KHTML, like Gecko) Version/5.0.2 Mobile/8C148 Safari/6533.18.5`

**Samsung Galaxy Note 3:** `Mozilla/5.0 (Linux; U; Android 4.3; en-us; SM-N900T Build/JSS15J) AppleWebKit/534.30 (KHTML, like Gecko) Version/4.0 Mobile Safari/534.30`

**Samsung Galaxy Note II:** `Mozilla/5.0 (Linux; U; Android 4.1; en-us; GT-N7100 Build/JRO03C) AppleWebKit/534.30 (KHTML, like Gecko) Version/4.0 Mobile Safari/534.30`

**Samsung Galaxy S III:** `Mozilla/5.0 (Linux; U; Android 4.0; en-us; GT-I9300 Build/IMM76D) AppleWebKit/534.30 (KHTML, like Gecko) Version/4.0 Mobile Safari/534.30`

Also, as @kumarharsh already noticed, the IE 10 user agent also seems to raise an issue (User Agent I used: `Mozilla/5.0 (compatible; MSIE 10.0; Windows NT 6.1; Trident/6.0`)

The only thing all these user agents have in common is the Mozilla/5.0 part, but that doesn't seem to be the problem, as you can just switch the "MSIE 10.0" part with "MSIE 11.0" and it magically works again.

I haven't tested any of these with the actual devices, but as far as I can tell the issue would be the same. This is actually breaking a lot of things, please fix :)

(Not sure if that's also a cause for the issue, but I'm using Meteor 1.3-beta.11, React 0.14.7 and Radium 0.16.6)

**Comment by rynti** [comment]
Okay, just figured out that this issue is actually coming from [inline-style-prefixer](https://github.com/rofrischmann/inline-style-prefixer).

I've opened an issue there: https://github.com/rofrischmann/inline-style-prefixer/issues/65

I suggest we keep this issue open until it's fixed on their end so we can update radium to use the updated inline-style-prefixer version in order to fix it here as well.

**Comment by rynti** [comment]
[inline-style-prefixer](https://github.com/rofrischmann/inline-style-prefixer) has just been updated to 1.0.0, fixing this issue as soon as Radium updates it's inline-style-prefixer version. (See #609)

**Comment by treylon** [comment]
Any updates on this?

**Comment by johanneslumpe** [comment]
@alexlande is there any chance that this can happen anytime soon? For us this is currently a blocker, as we need to apply quite a few `display: inline-block` styles to elements.

**Comment by alexlande** [comment]
Hmm, this isn't as straightforward as I had hoped. `inline-style-prefixer` 1.0 requires a `Set` polyfill (see https://github.com/rofrischmann/inline-style-prefixer/issues/73#issuecomment-195239581).

This would be a breaking change for Radium for the sake of one bug fix, which is unfortunate. There's also the fact that a `Set` polyfill will add additional page weight. https://www.npmjs.com/package/es6-shim comes out to about 4.5kb gzipped, which is a ton. There may be a smaller option out there that we can recommend, but I'm not sure what it is.

That said, this is a pretty severe bug, so we may need to bite the bullet here.

/cc @ianobermiller @tptee

**Comment by rofrischmann** [comment]
I was aware that using `Set` most likely will cause some pain lately... That said I don't think it's wise to step back to basic ES5 features just because this is not polyfilled by Babel and still not supported in every browser.
I can right now recommend using [es6-set](https://github.com/medikoo/es6-set) as a polyfill which seems to be quite lightweight and still ships everything we need, but as far as I've learned it is neither my nor your job to provide the polyfill as the user himself should add this if his targeted environment does not support `Set`. 

Still as it is kind of a small improvement (if at all) I could revert the move to `Set`. It just seems to be more "right" to do sth. like
 ```javascript
const properties = new Set([
    'prop1',
    'prop2'
])

if (properties.has(property)) {}
```
instead of 
```javascript
const properties = {
   prop1: true,
   prop2: true
}

if (properties(property)) {}
```

**Comment by alexlande** [comment]
Definitely agree that the usage of `Set` is nice here. This is a hard one. If we were talking about a React-specific library (like Radium), I would expect that library to match the same browser support matrix as React itself, meaning that the library wouldn't require additional polyfills that React itself doesn't (unless it was critical for the usage of the library, of course).

`inline-style-prefixer` isn't tied to any particular JS ecosystem (although it's probably fair to say that it's mostly used in React projects), so it's more difficult to draw that line.

For what it's worth, `es6-set` is larger than `inline-style-prefix-all`, and between it and Radium we would be costing users `20kb` of JS bundle size.

**Comment by johanneslumpe** [comment]
@rofrischmann While being able to use `Set` is nice, adding a huge overhead for users for something which can be easily emulated (per your example above) with wide-spread tools seems not justifiable just for the sake of using ES6 IMHO. If this is the only thing that's keeping radium from using 1.0 I'd be really happy if you were to make a decision to revert the usage of `Set` here.

**Comment by rynti** [comment]
I have to agree with @johanneslumpe here, and as far as I was able to see sets are only used for their `has` method in inline-style-prefixer, which can very easily be emulated with an object or array in case there's no Set implemented. I think that's the way to go, especially since this makes it easier to use your library from a developer's perspective.

**Comment by rofrischmann** [comment]
You're right pointing out that it is a huge overload for just this very special use case. Still it can be considered more performant and might even get more performant in the future. I guess this is sth. we want to think about again in a couple of years when ES2015 finally landed support everywhere.

I agree to reset those changes. Though I do not have time until next saturday to actually do this. If someone would take some minutes to apply those changes to **both** inline-style-prefixer and **inline-style-prefix-all** (pretty much the same files) I could perhaps merge and release on the go.
(Giving my mac into repair)

**Comment by alexlande** [comment]
Thanks @rofrischmann! I'll see if I can find some time today or tomorrow to submit a couple of PRs.

**Comment by johanneslumpe** [comment]
@alexlande Are there other breaking changes regarding the 1.0 of the prefixer? (We have to launch something today so I'm inclined to just create a fork of radium and include the set prefix just to get our stuff out)

**Comment by alexlande** [comment]
@johanneslumpe: Nope, that's the only breaking change that I'm aware of.

**Comment by johanneslumpe** [comment]
@alexlande awesome. I wanted to do some prs for the prefixer, but I won't be able to do so until tomorrow/ Friday (eu time). Guess I will try the fork now ;)

**Comment by alexlande** [comment]
:+1: Sorry for the trouble on this one! Will try to get it sorted out as soon as possible.

**Comment by johanneslumpe** [comment]
@alexlande No worries! I just noticed a weird thing though: installing from the forked repo directly does not include the `src` dir and thus the `postinstall` hook fails. Did you encounter this before?

**Comment by alexlande** [comment]
@johanneslumpe: Not sure about that one, sorry. Everyone else: this is fixed in Radium 0.17.0. Thanks for the help all!

**Comment by johanneslumpe** [comment]
@alexlande great, thanks for the quick fix!
